### PR TITLE
feat: support for Windows

### PR DESCRIPTION
**Pull request by ii64**
* fix (loader): default loader now has build tag: for linux and darwin
* feat (loader): add memory allocator support for Windows

### CODE REVIEW

**Review by AsterDY — COMMENTED** [review]
please use original format:
1. indent: 'four space' instead of 'tab'
2. import quote: `` instead of ""

**Comment by ii64** [comment]
Fixed

**Review by AsterDY — COMMENTED** [review]
Could you please add a CI file (ex: https://github.com/bytedance/sonic/blob/main/.github/workflows/push-check-go117.yml) to test if it works? use 'windows-latest' as basic image maybe?

File: internal/loader/loader_windows_amd64.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/*
```
[the remark this answers is not in the document]
Comment (ii64) [reply internal/loader/loader_windows_amd64.go]: Not neccessary, i've renamed it to `loader_windows.go` now.

**Comment by AsterDY** [comment]
It seems like go racer's problem: https://github.com/golang/go/issues/46099, and not gets fixed yet. Maybe removing `-race` test flag is feasible solution

**Comment by ii64** [comment]
Ouch, the GitHub Windows CI host got OOM, [it](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) has 7 GB RAM. I can reproduce TSAN error on my machine with `-race` flag, removing it fixed the problem. On issue_test, using -race took ~20x more RAM usage, could it might pass the test if memory/page were still available?


Attached the ut logs:
[ut_norace_ok.log](https://github.com/bytedance/sonic/files/8740008/ut_norace_ok.log)
[ut_race_err_tsan.log](https://github.com/bytedance/sonic/files/8740010/ut_race_err_tsan.log)
[ut_race_err_tsan_2.log](https://github.com/bytedance/sonic/files/8740013/ut_race_err_tsan_2.log)

File: .github/workflows/push-check-go-windows.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Unit Test
+        run: go test -v -gcflags -d=checkptr=0 -covermode atomic -coverprofile coverage.out ./...
+        env:
+          GOMAXPROCS: 4
+
+      - name: Generic Test
+        run: go test -v -gcflags -d=checkptr=0 -covermode atomic ./generic_test
```
Comment (AsterDY) [inline .github/workflows/push-check-go-windows.yml:26]: typo: `-gcflags -d=checkptr=0` should be `-gcflags=-d=checkptr=0`

**Review by AsterDY — APPROVED** [review]

**Comment by AsterDY** [comment]
Thanks to @ii64! This is the first effective PR from an outside developer! 🎉

**Pull request merged**